### PR TITLE
helm3: Migrate from dep to go modules

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/roadmap/issues/30

We now have a `helm3` branch that we will use for the rest of the migration. This is based on https://github.com/giantswarm/chart-operator/pull/351 from Pawel.

Once the branch is using modules I'll introduce the helm3 branch of helmclient.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	k8s.io/kube-controller-manager v0.0.0 => k8s.io/kube-controller-manager v0.16.6
+	k8s.io/kube-proxy v0.0.0 => k8s.io/kube-proxy v0.16.6
+	k8s.io/kube-scheduler v0.0.0 => k8s.io/kube-scheduler v0.16.6
+	k8s.io/kubectl v0.0.0 => k8s.io/kubectl v0.16.6
+	k8s.io/kubelet v0.0.0 => k8s.io/kubelet v0.16.6
+	k8s.io/legacy-cloud-providers v0.0.0 => k8s.io/legacy-cloud-providers v0.16.6
+	k8s.io/metrics v0.0.0 => k8s.io/metrics v0.16.6
+	k8s.io/sample-apiserver v0.0.0 => k8s.io/sample-apiserver v0.16.6
+	k8s.io/utils v0.0.0 => k8s.io/utils v0.0.0-20191114200735-6ca3b61696b6
+)
```
Comment (kopiczko) [inline go.mod]: Do we need this replace BS when we don't have dependency on k8s.io/kubernetes?

Comment (rossf7) [reply go.mod]: I can check but this is just an intermediate step before adding the helm3 branch.

The dependencies then get radically simpler. See https://github.com/giantswarm/helmclient/pull/178.

Comment (kopiczko) [reply go.mod]: Ack. Ignore me then.

Comment (rossf7) [reply go.mod]: Yes situation will improve soon. Going with this.

**Pull request merged**